### PR TITLE
Bump ``types-docutils`` to 0.21.0.20240711

### PR DESCRIPTION
**Pull request by Carreau**
Bumps [types-docutils](https://github.com/python/typeshed) from 0.21.0.20240708 to 0.21.0.20240710 

Same as #12526, but in addition, fix the types annotations.

### CODE REVIEW

File: sphinx/directives/__init__.py [hunk 1]
```diff
@@ -316,16 +317,22 @@ def run(self) -> list[Node]:
             docutils.unregister_role('')
             return []
         role_name = self.arguments[0]
-        role, messages = roles.role(role_name, self.state_machine.language,
+        messages: list[SystemMessage | system_message] = []
+        messages_: list[SystemMessage]
```
Comment (AA-Turner) [inline sphinx/directives/__init__.py]: ``SystemMessage`` is a Docutils exception type, only raised on warnings with severity greater than ``halt_level`` -- it shouldn't be returned as a node type?

Comment (Carreau) [reply sphinx/directives/__init__.py]: This is what the new docutils types are saying. I can check upstream.

Comment (AA-Turner) [reply sphinx/directives/__init__.py]: It looks like the following should be ``nodes.system_message`` instead:

https://github.com/python/typeshed/blob/57c524d8225faae98611e6b643bb29769afdb3fa/stubs/docutils/docutils/parsers/rst/roles.pyi#L21-L23

https://github.com/python/typeshed/blob/57c524d8225faae98611e6b643bb29769afdb3fa/stubs/docutils/docutils/parsers/rst/directives/__init__.pyi#L12-L14

That has been wrong for two years, but perhaps something in @danieleades' latest change made MyPy look at those functions?

A

Comment (Carreau) [reply sphinx/directives/__init__.py]: Yep, came to the same conclusion. will send update to typeshed.

Comment (Carreau) [reply sphinx/directives/__init__.py]: See https://github.com/python/typeshed/pull/12314

**Comment by AA-Turner** [comment]
Thanks @Carreau!

A

**Pull request merged**